src/lib/IO/IOMetadataHandler/Flysystem.php

```diff
@@ -56,6 +68,12 @@ public function exists($spiBinaryFileId): bool
     {
         try {
             return $this->filesystem->fileExists($spiBinaryFileId);
+        } catch (CorruptedPathDetected $e) {
+            $this->logger->error(
+                sprintf('Binary file with ID="%s" does not exist: %s', $spiBinaryFileId, $e->getMessage())
+            );
```

Comment:
Since we already caught it, we might as well attach it to the log.
```suggestion
            $this->logger->error(
                sprintf('Binary file with ID="%s" does not exist: %s', $spiBinaryFileId, $e->getMessage()),
                ['exception' => $e],
            );
```